cmd/siklu/common_boot.c

```diff
@@ -63,22 +86,44 @@ int load_kernel_image(void) {
 	int ret;
 	char formatted_bootargs[1024];
 	const char *old_bootargs;
+	char *boot_cmd_format;
 	unsigned int fdt_addr;
-
-	if (strict_strtoul(dtb_load_address(), 16, &fdt_addr) < 0)
-		return -EINVAL;
-
-	const char* fdt_param = siklu_fdt_getprop_string(fdt_addr, "/chosen", "bootargs", NULL);
-
-	if (!IS_ERR(fdt_param)) {
-		old_bootargs = env_get("bootargs");
-		snprintf(formatted_bootargs, sizeof(formatted_bootargs), "%s %s", old_bootargs, fdt_param);
-		printf("SIKLU BOOT: Added DTS-specific bootargs: %s\n", fdt_param);
-		env_set("bootargs", formatted_bootargs);
+	char *fdt_file;
+	char *vendor;
+
+	if (is_fit_image()) {
+		fdt_file = env_get("fdtfile");
+		if (fdt_file == NULL) {
+			printk(KERN_ERR "Could not find FDT file environment variable");
+			return -EINVAL;
+		}
+		vendor = env_get("vendor");
+		if (vendor == NULL) {
+			printk(KERN_ERR "Could not find FDT file environment variable");
+			return -EINVAL;
+		}
+		if (*vendor >= 'A' && *vendor <= 'Z')
+			*vendor += 'a' - 'A';
```

Comment:
what are we doing here?

Reply:
We need the "marvell" vendor prefix for the conf but the env variable is "Marvell" so for it not to be hard-coded I just lower-cased the first letter

Reply:
I don't think that it is a good assumption to make. vendor has no direct connection with the vendor in the fdt and it be changed in other versions.  I would suggest to remove the vendor from the code that generates the its for the fitimage.